### PR TITLE
Drag&drop improvements & some minor fixes

### PR DESCRIPTION
**Pull request by rpominov**
- #175

## Before requesting a review

- [x] if the PR is WIP - use draft mode
- [x] made a self-review
- [x] added inline comments where things may be not obvious (the "why", not "what")
- [x] what kind of review is needed?
  - [x] conceptual
  - [ ] detailed
  - [ ] with testing

## Test cases

- [ ] step by step interaction description and what is expected to happen

## Before merging

- [x] tested locally and on preview environment (preview dev login: 5de6)
- [x] updated [test cases](https://github.com/webstudio-is/webstudio-designer/blob/main/apps/designer/docs/test-cases.md) document
- [x] added tests
- [x] if any new env variables are added, added them to `.env.example` and the `designer/env-check.js` if mandatory

### CODE REVIEW

File: packages/design-system/src/components/text.tsx [hunk 1]
```diff
@@ -57,16 +57,17 @@ export const Text = styled("div", {
       true: {
         whiteSpace: "nowrap",
         textOverflow: "ellipsis",
+        overflow: "hidden",
+
+        // To make sure text is not clipped vertically
+        pt: 10,
```
Comment (kof) [inline packages/design-system/src/components/text.tsx:63]: we probably don't need 10px, more like 3-5, but also don't know if that matters in any way

Comment (kof) [reply packages/design-system/src/components/text.tsx:63]: oh wait, in px it means its a fixed size so a larger font won't work with this, wouldn't it make sense to use em?

File: packages/design-system/src/components/primitives/dnd/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+This division of responsibilities between the two hooks allows to put them in different windows. For example, `useDrag` can be instantiated in the main browser window, while `useDrop` can live in an iframe. The information you’ll need to pass between them can be transferred via `postMessage()`.
+
+| Needs access to: | Drop targets DOM | Drag items DOM | Scroll container |
+| ---------------- | ---------------- | -------------- | ---------------- |
+| `useDrag`        |                  | V              |                  |
+| `useDrop`        | V                |                | V                |
+| `useAutoScroll`  |                  |                | V                |
+| `useHold`        |                  |                |                  |
+| `useDragCursor`  |                  |                |                  |
```
Comment (kof) [inline packages/design-system/src/components/primitives/dnd/README.md:49]: Well done readme 🔥

**Review by kof — APPROVED** [review]
looking great 🎉

File: packages/design-system/src/components/primitives/dnd/use-drop.ts [hunk 3]
```diff
@@ -219,40 +220,44 @@ export const useDrop = <Data>(props: UseDropProps<Data>): UseDropHandlers => {
...
+            candidate.element.getBoundingClientRect()
+          ),
+        };
+      };
+
       // @todo: Cache this?
       // Not expensive by itself, but it may call elementToData multiple times.
-      let candidate =
+      let candidate = withArea(
         root &&
```
[the remark this answers is not in the document]
Comment (rpominov) [reply packages/design-system/src/components/primitives/dnd/use-drop.ts:242]: it won't do anything when called with `null/undefined`

**Pull request merged**